test/port-finder-test.js

```diff
@@ -46,6 +46,35 @@ vows.describe('portfinder').addBatch({
           assert.isTrue(!err);
           assert.equal(port, 32773);
         }
+      },
+      "the getPort() method with stopPort smaller than available port": {
+        topic: function() {
+          // stopPort: 32722 is smaller than available port 32773 (32768 + 5)
+          setTimeout(function() {
+            portfinder.getPort({ stopPort: 32772 }, this.callback);
+          }.bind(this), 3000); //wait for cleanup of bound hosts.
+        },
+        "should return error": function(err, port) {
+          assert.isTrue(!!err);
+          assert.equal(
+            err.message,
+            'No open ports found in between 32768 and 32772'
+          );
+          return;
+        }
+      },
+      "the getPort() method with stopPort greater than available port": {
+        topic: function() {
+          // stopPort: 32774 is greater than available port 32773 (32768 + 5)
+          setTimeout(function() {
+            portfinder.getPort({ stopPort: 32774 }, this.callback);
+          }.bind(this), 3000); //wait for cleanup of bound hosts.
+        },
+        "should respond with the first free port (32773) less than provided stopPort": function(err, port) {
+          if (err) { debugVows(err); }
+          assert.isTrue(!err);
+          assert.equal(port, 32773);
+        }
```

Comment:
i think the socket is not fully closed by the time these are running for port 32773 - that's what those setTimeouts of 3 seconds are for -- its hacky i know - if you know vows and have a better way, by all means... 

it passes locally i assume?

Reply:
also check https://github.com/indexzero/node-portfinder/blob/master/test/helper.js - we start some httpServer's in there, might be messing with your stopPort

I can't recall, but will look deeper before we merge for sure

Reply:
Yes, they are all passing locally.

Reply:
Yeah. you are right. All the tests were run asynchronously when test servers are spun up and there is a race condition in the tests. I think the way to fix this is to have all the tests spun their own test services which are not colliding. I'd like to do this as a followup.

Reply:
>  I think the way to fix this is to have all the tests spun their own test services which are not colliding. I'd like to do this as a followup.

agreed, and sounds awesome, thanks